sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -133,12 +133,46 @@ class AstBuilder extends DataTypeAstBuilder with SQLConfHelper with Logging {
 
   private def visitCompoundBodyImpl(
       ctx: CompoundBodyContext,
-      label: Option[String]): CompoundBody = {
+      label: Option[String],
+      allowVarDeclare: Boolean): CompoundBody = {
     val buff = ListBuffer[CompoundPlanStatement]()
     ctx.compoundStatements.forEach(compoundStatement => {
       buff += visit(compoundStatement).asInstanceOf[CompoundPlanStatement]
     })
 
+    val compoundStatements = buff.toList
+
+    if (allowVarDeclare) {
+      val declareAfterPrefix = compoundStatements
+        .dropWhile(statement => statement.isInstanceOf[SingleStatement] &&
+          statement.asInstanceOf[SingleStatement].parsedPlan.isInstanceOf[CreateVariable])
+        .filter(_.isInstanceOf[SingleStatement])
+        .find(_.asInstanceOf[SingleStatement].parsedPlan.isInstanceOf[CreateVariable])
+
+      declareAfterPrefix match {
+        case Some(SingleStatement(parsedPlan)) =>
+          throw SqlScriptingErrors.variableDeclarationOnlyAtBeginning(
+            parsedPlan.asInstanceOf[CreateVariable].name.
+              asInstanceOf[UnresolvedIdentifier].nameParts.last,
+            parsedPlan.origin.line.get.toString)
+        case _ =>
+      }
+
+    } else {
+      val declare = compoundStatements
+        .filter(_.isInstanceOf[SingleStatement])
+        .find(_.asInstanceOf[SingleStatement].parsedPlan.isInstanceOf[CreateVariable])
+
+      declare match {
+        case Some(SingleStatement(parsedPlan)) =>
+          throw SqlScriptingErrors.variableDeclarationOnlyAtBeginning(
+            parsedPlan.asInstanceOf[CreateVariable].name.
+              asInstanceOf[UnresolvedIdentifier].nameParts.last,
```

Comment:
when I split one chained method calls into multiple lines, I usually like to start the line with the . because it's then clear at the first glance that it is continuation from the row before... also, I try to group them logically.
so, I would rather do:
```
parsedPlan.asInstanceOf[CreateVariable]
    .name.asInstanceOf[UnresolvedIdentifier]
    .nameParts.last
```

same goes for the similar part of the code above.